src/byte_set.rs

```diff
@@ -126,21 +133,55 @@ impl<const LEN: usize> From<[u8; LEN]> for ByteSet {
     }
 }
 
-impl From<Range<u8>> for ByteSet {
-    fn from(range: Range<u8>) -> ByteSet {
+impl<'a> From<&'a [u8]> for ByteSet {
+    fn from(bytes: &'a [u8]) -> ByteSet {
         let mut bs = ByteSet::new();
-        for b in range {
-            bs.insert(b);
+        for b in bytes {
+            bs.insert(*b);
         }
         bs
     }
 }
 
-impl From<RangeInclusive<u8>> for ByteSet {
-    fn from(range: RangeInclusive<u8>) -> ByteSet {
-        let mut bs = ByteSet::new();
-        for b in range {
-            bs.insert(b);
+impl From<ops::Range<u8>> for ByteSet {
+    fn from(value: ops::Range<u8>) -> Self {
+        ByteSet::from(value.start..=value.end - 1)
```

Comment:
Will this behave on empty ranges? In particular 0..0 :thinking:

Reply:
Indeed, there needs to be an edge-case to handle such ranges.

Reply:
Fixed this in the latest commit.